src/compiler/crystal/semantic/math_interpreter.cr

```diff
@@ -6,24 +6,28 @@ struct Crystal::MathInterpreter
   def initialize(@path_lookup : Type, @visitor : SemanticVisitor? = nil, @target_type : IntegerType? = nil)
   end
 
+  private def make_error(node, message)
+    Crystal::TypeException.for_node(node, message)
+  end
```

Comment:
Not sure if this abstraction is really much use.
Just mentioning this in case sombody agrees. It's not a call for action.